src/App.js

```diff
@@ -4,10 +4,28 @@ import chatMessages from './data/messages.json';
 import ChatLog from './components/ChatLog';
 
 const App = () => {
-  const [totalLikes, setTotalLikes] = useState(0); 
+  const [messages, setMessages] = useState(chatMessages);
 
-  const handleLikeCount = (liked) => {
-    setTotalLikes((totalLikes) => liked ? totalLikes + 1 : totalLikes - 1); 
+  const calculateTotalLikes = (messages) => {
+    return messages.reduce((totalLikes, message) => {
+      if (message.liked) {
+        return totalLikes + 1; 
+      }
+      return totalLikes;
+    }, 0); 
+  };
+  
+  const totalLikes = calculateTotalLikes(messages);
+
+  const handleLikeClicked = (id) => {
+    const newMessages = messages.map(message => {
+      if (message.id === id) {
+        return {...message, liked: !message.liked};
+      } else {
+        return message;
+      }
+    });
+    setMessages(newMessages);
   };
```

Comment:
Great use of `map` and spreading to make a new array, and a new modified message. The one recommendation I would make is that since the next value of `messages` depends on the current, we can write the set call using the callback style. It's not _really_ a problem here, since there's nothing asynchronous (the most common way we'd see an issue), but it doesn't hurt to get in the habit.

```js
  const handleLikeClicked = (id) => {
    setMessages(messages => (
      messages.map(message => {
        if (message.id === id) {
          return {...message, liked: !message.liked};
        } else {
          return message;
        }
      })
    ));
  };
```